### PR TITLE
cmd/snap-bootstrap/initramfs-mounts: verify the seed snap matches bootloader env

### PR DESCRIPTION
**Pull request by anonymouse64**
The recovery system has a boot env which contains what specific kernel should be run/mounted in the snapd_recovery_kernel env var and this is supposedly what kernel was booted to get us to install mode, but we should verify this matches. Also add unit tests for the happy path and common unhappy paths.

This includes https://github.com/snapcore/snapd/pull/8365

Also some drive-bys:
* delete the cross-check TODO, as recoverySystemEssentialSnaps already is verifying the snap's signatures are valid and this is probably sufficient checks for install mode.
* fix variable name in makebootable to match what it stands for
* add an Info() method to seed.Snap (if this is controversial I can back this out)
* delete unnecessary TODO, there isn't a better way to mount ubuntu-data as tmpfs without specifying the options to systemd-mount here right now

### CODE REVIEW

File: seed/seed.go [hunk 1]
```diff
@@ -61,6 +61,10 @@ func (s *Snap) ID() string {
 	return s.SideInfo.SnapID
 }
 
+func (s *Snap) Info() snap.Info {
```
Comment (pedronis) [inline seed/seed.go]: if this returned a snap.PlaceInfo it would be less misleading

Comment (anonymouse64) [reply seed/seed.go]: sure, that's easy to change, also stop working on your day off! :smiley:

Comment (pedronis) [reply seed/seed.go]: it's all kind of gdocs fault :smile:

**Comment by anonymouse64** [comment]
Yet again this is failing to boot 100% of the time on GCE but works fine locally with QEMU :confused:

**Comment by pedronis** [comment]
@anonymouse64 let's chat about this vs next steps in the standup

**Comment by pedronis** [comment]
this is doing what was envisioned, but it's a bit unclear how much value is adding right away, especially because it has some cost in terms of added code (function in function) in bootstrap, I asked @anonymouse64 to focus on more tangible checks for beta and I'm marking this one blocked for now

**Comment by anonymouse64** [comment]
@pedronis should we consider this for 1.0?

**Comment by anonymouse64** [comment]
Closing until we have a better way to tell what kernel we actually booted

**Pull request closed without merging**